metadata-ingestion/tests/unit/test_transform_dataset.py

```diff
@@ -2841,3 +2844,188 @@ def test_add_dataset_data_product_transformation():
+                models.OwnerClass(owner=owner, type=models.OwnershipTypeClass.DATAOWNER)
+                for owner in in_owners
+            ]
+        )
+
+    in_pipeline_context.graph.get_ownership = fake_ownership_class  # type: ignore
+
+    output = run_dataset_transformer_pipeline(
+        transformer_type=PatternCleanUpOwnership,
+        aspect=models.OwnershipClass(
+            owners=[
+                models.OwnerClass(owner=owner, type=models.OwnershipTypeClass.DATAOWNER)
+                for owner in in_owners
+            ]
+        ),
+        config={"pattern_for_cleanup": config},
+        pipeline_context=in_pipeline_context,
+    )
+
+    assert len(output) == 2
+    ownership_aspect = output[0].record.aspect
+    assert isinstance(ownership_aspect, OwnershipClass)
+    assert len(ownership_aspect.owners) == len(in_owners)
+
+    out_owners = [owner.owner for owner in ownership_aspect.owners]
+    assert set(out_owners) == set(cleaned_owner_urn)
+
+
+def test_clean_owner_urn_transformation(mock_datahub_graph):
+    pipeline_context = PipelineContext(run_id="transformer_pipe_line")
+    pipeline_context.graph = mock_datahub_graph(DatahubClientConfig())
+
+    user_emails = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:123email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id1@example.com",
+        "email_id:id2@example.com",
+    ]
+
+    in_owner_urns: List[str] = []
+    for user in user_emails:
+        in_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+
+    # remove ':ABCDEF'
+    config: List[Union[re.Pattern, str]] = [":ABCDEF"]
+    expected_user_emails: List[str] = [
+        "email_id@example.com",
+        "123email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id1@example.com",
+        "email_id:id2@example.com",
+    ]
+    expected_owner_urns: List[str] = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+    # remove multiple values
+    config = [":ABCDEF", "email"]
+    expected_user_emails = [
+        "_id@example.com",
+        "123_id@example.com",
+        "_id@example.co.in",
+        "_id@example.co.uk",
+        "_test:XYZ@example.com",
+        "_id:id1@example.com",
+        "_id:id2@example.com",
+    ]
+    expected_owner_urns = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+    # remove values using RegEx
```

Comment:
What values is this regex supposed to be removing? Please clarify this regex.

Reply:
addressed and updated the comment